### ISSUE TITLE
the checker says "it looks all good", but "test.sh: 3: [: Illegal number:" error is reported when being executed

### ISSUE THREAD

**Issue by Oscarzhao** [koalaman/shellcheck#474]
```
#!/bin/sh

if [ "$(ifconfig docr0 >/dev/null 2>&1)" -eq 0 ]; then
        echo "---==="
fi
if [ -z "dddd" ]; then
        echo "eee"
fi
```

**Comment by deryni** [comment]
Are you expecting shellcheck to attempt to divine whether the command in your `$(...)` substitution is going to return a value that is valid for the `[ ...]` test it is in here?

**Comment by kurahaupo** [comment]
And then maybe add a warning like
> empty string is not valid for numeric comparison in ```[ NUM -eq NUM ]```

**Comment by deryni** [comment]
In this sort of case, true, that would be a useful additional warning. In general though that's only going to be possible for literal strings in tests (because tracking contents is a more complicated task that I don't think shellcheck is really in a position to do, though that sort of static analysis would be pretty cool).
